Fabric/build.gradle

```diff
@@ -29,21 +33,98 @@ loom {
             ideConfigGenerated(true)
             runDir("run")
         }
+        // This adds a new gradle task that runs the datagen API: "gradlew runDatagen"
+        datagen {
+            inherit server
+            name "Data Generation"
+            setConfigName("Fabric Datagen")
+            ideConfigGenerated(true)
+            vmArg "-Dfabric-api.datagen"
+            vmArg "-Dfabric-api.datagen.output-dir=${file("src/main/generated")}"
+            vmArg "-Dfabric-api.datagen.modid=${mod_id}"
+ 
+            runDir "build/datagen"
+        }
     }
 }
 
+// Add the generated resources to the main source set
+sourceSets {
+    main {
+        resources {
+            srcDirs += [
+                    'src/main/generated'
+            ]
+        }
+    }
+}
+
+dependencies {
+	// To change the versions see the gradle.properties file
+	minecraft "com.mojang:minecraft:${minecraft_version}"
+
+    switch (mappings_type) {
+        case "official" :
+            mappings loom.officialMojangMappings()
+            break
+        case "parchment" :
+            mappings loom.layered() {
+                officialMojangMappings()
+                parchment("org.parchmentmc.data:parchment-${parchment_mc_version}:${parchment_mappings}@zip")
+            }
+            break
+        case "yarn" :
+            mappings "net.fabricmc:yarn:${minecraft_version}+${yarn_mappings}:v2"
+            break
+        case "quilt" :
+            mappings "org.quiltmc:quilt-mappings:$minecraft_version+$quilt_mappings_version:intermediary-v2"
+            break
+        case "quilted_official" :
+            mappings loom.layered {
+                mappings "org.quiltmc:quilt-mappings:$minecraft_version+$quilt_mappings_version:intermediary-v2"
+                officialMojangMappings()
+            }
+            break
+        default :
+            mappings loom.officialMojangMappings()
+            break
+    }
```

Comment:
This while section of code is never going to work until VanillaGradle supports different mapping types besides mojmap, and I don't believe that will be happening anytime soon.

Reply:
This is a draft, I was looking to adding loom to common project, as described in #13

Reply:
We have no plans to change the common project to Loom.

VanillaGradle just works, I have gotten reports that when people try and use loom, it may work for a bit but they always run into issues, but we have had no issues using VanillaGradle.